server-spi-private/src/main/java/org/keycloak/provider/EnvironmentDependentProviderFactory.java

```diff
@@ -28,22 +28,12 @@
 public interface EnvironmentDependentProviderFactory {
 
     /**
-     * @return <code>true</code> if the provider is supported and should be available, <code>false</code> otherwise
-     * @deprecated Prefer overriding/using the {@link #isSupported(Config.Scope)} method.
-     */
-    @Deprecated
-    default boolean isSupported() {
-        return false;
-    }
-
-    /**
-     * An alternative to {@link #isSupported()} method to check if the provider is supported based on the
-     * provider configuration.
+     * Check if the provider is supported and should be available based on the provider configuration.
      *
      * @param config the provider configuration
      * @return {@code true} if the provider is supported. Otherwise, {@code false}.
      */
     default boolean isSupported(Config.Scope config) {
-        return isSupported();
+        return false;
```

Comment:
IMHO there shouldn't be a default impl here any more, as there is now only a single method in the Interface. Either you implement the interface, or not. 

So please remove the default implementation.